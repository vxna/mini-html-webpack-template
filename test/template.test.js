const compiler = require('./helpers/compiler')
const getConfig = require('./helpers/config')

test('options: common', async () => {
  const config = getConfig({
    options: {
      title: 'common options',
      favicon: '/favicon.ico',
      container: 'root',
    },
  })

  const stats = await compiler('base.js', config)
  const asset = stats.compilation.assets['index.html']._value

  expect(asset).toMatchSnapshot()
})

test('options: advanced', async () => {
  const config = getConfig({
    options: {
      lang: 'en',
      title: 'advanced options',
      head: {
        meta: [
          {
            name: 'description',
            content: 'mini-html-webpack-template',
          },
          {
            property: 'og:description',
            content: 'mini-html-webpack-template',
          },
        ],
        links: [
          {
            rel: 'icon',
            type: 'image/x-icon',
            href: '/favicon.ico',
          },
        ],
        scripts: [
          {
            defer: '',
            type: 'text/javascript',
            src: 'https://unpkg.com/random',
          },
        ],
        raw: '<style id="head-raw-string"></style>',
      },
      body: {
        raw: [
          '<script id="body-raw-array-1"></script>',
          '<script id="body-raw-array-2"></script>',
        ],
      },
    },
  })

  const stats = await compiler('base.js', config)
  const asset = stats.compilation.assets['index.html']._value

  expect(asset).toMatchSnapshot()
})

test('options: output attrs', async () => {
  const config = getConfig({
    options: {
      title: 'output attrs',
      favicon: '/favicon.ico',
      container: 'root',
      attrs: {
        js: { async: '', type: 'text/javascript' },
        css: { type: 'text/css' },
      },
    },
  })

  const stats = await compiler('base.js', config)
  const asset = stats.compilation.assets['index.html']._value

  expect(asset).toMatchSnapshot()
})

test('options: trim whitespace', async () => {
  const config = getConfig({
    options: {
      title: 'trim whitespace',
      favicon: '/favicon.ico',
      container: 'root',
      trimWhitespace: true,
    },
  })

  const stats = await compiler('base.js', config)
  const asset = stats.compilation.assets['index.html']._value

  expect(asset).toMatchSnapshot()
})
