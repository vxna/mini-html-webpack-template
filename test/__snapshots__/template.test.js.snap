// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options: advanced 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>advanced options</title>
    <meta name=\\"description\\" content=\\"mini-html-webpack-template\\">
    <meta property=\\"og:description\\" content=\\"mini-html-webpack-template\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\">
    <script defer=\\"\\" type=\\"text/javascript\\" src=\\"https://unpkg.com/random\\"></script>
    <style id=\\"head-raw-string\\"></style>
    <link rel=\\"stylesheet\\" href=\\"main.css\\">
  </head>
  <body>
    <script id=\\"body-raw-array-1\\"></script>
    <script id=\\"body-raw-array-2\\"></script>
    <script src=\\"main.js\\"></script>
  </body>
</html>"
`;

exports[`options: common 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>common options</title>
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\">
    <link rel=\\"stylesheet\\" href=\\"main.css\\">
  </head>
  <body>
    <div id=\\"root\\"></div>
    <script src=\\"main.js\\"></script>
  </body>
</html>"
`;

exports[`options: output attrs 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>output attrs</title>
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\">
    <link type=\\"text/css\\" rel=\\"stylesheet\\" href=\\"main.css\\">
  </head>
  <body>
    <div id=\\"root\\"></div>
    <script async=\\"\\" type=\\"text/javascript\\" src=\\"main.js\\"></script>
  </body>
</html>"
`;

exports[`options: trim whitespace 1`] = `"<!DOCTYPE html><html><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><title>trim whitespace</title><link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\"><link rel=\\"stylesheet\\" href=\\"main.css\\"></head><body><div id=\\"root\\"></div><script src=\\"main.js\\"></script></body></html>"`;
